internal/app/handlers.go

```diff
@@ -1,74 +1,153 @@
-	"github.com/go-chi/chi/v5"
+	"github.com/Longreader/go-shortener-url.git/internal/tools"
+	"github.com/go-chi/chi"
+	"github.com/sirupsen/logrus"
 )
 
-var Store = storage.New()
+func (h *Handler) IDGetHandler(w http.ResponseWriter, r *http.Request) {
+
+	logrus.Debug("IDGetHandler start")
+	defer logrus.Debug("IDGetHandler end")
 
-func IDGetHandler(w http.ResponseWriter, r *http.Request) {
-	// Handle 'GET' method
-	// income ID-shorl url
-	// outcome Redirect by full url
 	if r.Method != http.MethodGet {
 		http.Error(w, "Only GET request are allowed", http.StatusMethodNotAllowed)
 		return
 	}
 
+	// h.Store.GetAll()
+
 	id := chi.URLParam(r, "id")
 
+	logrus.Debugf("ID equal %s", id)
+
 	if id == "" {
-		http.Error(w, "Bad request", http.StatusBadRequest)
+		http.Error(w, "Bad request or miss id", http.StatusBadRequest)
 		return
 	}
 
-	// WORK with storage
-	// Move id - data url
-	// url, ok := storage[vars["id"]]
-	fullURL, ok := Store.Get(id)
+	fullURL, ok := h.Store.Get(id)
 
 	if !ok {
-		http.Error(w, "Bad request", http.StatusBadRequest)
+		http.Error(w, fmt.Sprintf("Wrong ID %s", fullURL), http.StatusBadRequest)
 		return
 	}
-	// log.Println("IDFefaultHandler func()\n\tOK is", ok, "\n\tItem is", fullURL)
 
-	w.Header().Add("Content-Type", "text/html; charset=UTF-8")
+	// if r.Header.Get("Accept-Encoding") == "gzip" {
+	// 	w.Header().Set("")
+	// }
 	w.Header().Set("Location", string(fullURL))
+	w.Header().Add("Content-Type", "text/html; charset=UTF-8")
 	w.WriteHeader(307)
 }
 
-func ShortenerURLHandler(w http.ResponseWriter, r *http.Request) {
-	// Handle POST method
-	// income full URL
-	// outcome None
-	// set up pair 'full url' - 'id' in storage
+func (h *Handler) ShortenerURLHandler(w http.ResponseWriter, r *http.Request) {
+
+	logrus.Debug("ShortenerURLHandler start")
+	defer logrus.Debug("ShortenerURLHandler end")
+	logrus.Debugf("Base URL %s", h.BaseURL)
+
 	if r.Method != http.MethodPost {
 		http.Error(w, "Bad request", http.StatusBadRequest)
 		return
 	}
-	// Read URL
-	fullURL, err := io.ReadAll(r.Body)
-	// Handle error
+
+	var reader io.Reader
+
+	if r.Header.Get(`Content-Encoding`) == `gzip` {
+		gz, err := gzip.NewReader(r.Body)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		reader = gz
+		defer gz.Close()
+	} else {
+		reader = r.Body
+	}
+
+	fullURL, err := io.ReadAll(reader)
+	defer r.Body.Close()
+
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
 	}
-	// WORK with full url
-	shortURL := shortener.RandStringBytes(7)
-	for _, ok := Store.Get(shortURL); ok; {
-		shortURL = shortener.RandStringBytes(7)
-		_, ok = Store.Get(shortURL)
+
+	shortURL := tools.RandStringBytes(7)
+	for _, ok := h.Store.Get(shortURL); ok; {
+		shortURL = tools.RandStringBytes(7)
+		_, ok = h.Store.Get(shortURL)
 	}
-	baseURL := "http://127.0.0.1:8080/"
-	// Write shor url to body
+
 	w.WriteHeader(201)
```

Comment:
```suggestion
	w.WriteHeader(http.StatusCreated)
```
И неплохо бы выставить Content-Type (text/plain)